python/delta/tables.py

```diff
@@ -96,7 +120,54 @@ def merge(self, source, condition):
         not. See :class:`DeltaMergeBuilder` for a full description of this operation and what
         combinations of update, delete and insert operations are allowed.
 
-        :return: a :py:class:`delta.tables.DeltaMergeBuilder` object
+        Example 1 with conditions and update expressions as SQL formatted string::
+
+            deltaTable.alias("events").merge(
+                source = updatesDF.alias("updates"),
+                condition = "events.eventId = updates.eventId"
+              ).whenMatchedUpdate(set =
+                {
+                  "data": "updates.data",
+                  "count": "events.count + 1"
+                }
+              ).whenNotMatchedInsert(values =
+                {
+                  "date": "updates.date",
+                  "eventId": "updates.eventId",
+                  "data": "updates.data",
+                  "count": "1"
+                }
+              ).execute()
+
+        Example 2 with conditions and update expressions as Spark SQL functions::
+
+            from pyspark.sql.functions import *
+
+            deltaTable.alias("events").merge(
+                source = updatesDF.alias("updates"),
+                condition = expr("events.eventId = updates.eventId")
+              ).whenMatchedUpdate(set =
+                {
+                  "data" : col("updates.data"),
+                  "count": col("events.count") + 1
+                }
+              ).whenNotMatchedInsert(values =
+                {
+                  "date": col("updates.date"),
+                  "eventId": col("updates.eventId"),
+                  "data": col("updates.data"),
+                  "count": lit("1")
+                }
+              ).execute()
+
+        :param source: Condition to match
+        :type source: str or pyspark.sql.Column
```

Comment:
Are the above two lines correct? Isn't it a DataFrame?

Reply:
my mistake. you are right.